### PR TITLE
Fix stencil issues in GL on thread

### PR DESCRIPTION
**Pull request by unknownbrackets**
This makes Star Ocean work correctly (sans thorough testing) on GLES.  As a bonus, it also fixes stencil on Vulkan too (which kinda worked, but didn't clear properly, so got dirty when you scrolled.)

Also fixes Valkyria Chronicles 3, which wasn't playable at all without the fix.  Very possible other games, I've definitely seen stencil only clears in several effects.

-[Unknown]

### CODE REVIEW

File: GPU/Vulkan/StencilBufferVulkan.cpp [hunk 1]
```diff
@@ -150,26 +138,39 @@ bool FramebufferManagerVulkan::NotifyStencilUpload(u32 addr, int size, bool skip
 
 	VkDescriptorSet descSet = vulkan2D_->GetDescriptorSet(overrideImageView_, nearestSampler_, VK_NULL_HANDLE, VK_NULL_HANDLE);
 
+	if (usedBits == 0) {
+		// Note: Even with skipZero, we don't necessarily start framebuffers at 0 in Vulkan.  Clear anyway.
```
Comment (hrydgard) [inline GPU/Vulkan/StencilBufferVulkan.cpp]: Nice, this is neat.

**Review by hrydgard — APPROVED** [review]

File: GPU/Vulkan/StencilBufferVulkan.cpp (hunk 1 above)
Comment (unknownbrackets) [reply GPU/Vulkan/StencilBufferVulkan.cpp]: Hmm, seeing it again, maybe `value` needs to be 0xFF so the shader won't discard it, d'oh.

I guess parts where the upload has zero are still a problem though, so the skipZero case probably won't work thinking this through....

-[Unknown]

Comment (unknownbrackets) [reply GPU/Vulkan/StencilBufferVulkan.cpp]: Maybe `&& stencilValue != 0`, I mean, it's a constant so shouldn't have any perf impact...

-[Unknown]

Comment (hrydgard) [reply GPU/Vulkan/StencilBufferVulkan.cpp]: Hm.. yeah. Or could always add another push constant to the shader to change modes...

**Pull request merged**